packages/style-engine/src/styles/border/index.ts

```diff
@@ -60,7 +107,32 @@ const borderStyle: StyleDefinition = {
 
 const width: StyleDefinition = {
 	name: 'width',
-	generate: createBorderGenerateFunction( [ 'border', 'width' ] ),
+	generate: ( style: Style, options: StyleOptions ) => {
+		const borderWidthGenerate = createBorderGenerateFunction( [
+			'border',
+			'width',
+		] );
+		const borderStyleGenerate = createBorderGenerateFunction( [
+			'border',
+			'style',
+		] );
+		const output = borderWidthGenerate( style, options );
+		if (
+			style?.border?.width &&
+			! style?.border.color && // this is only here to avoid duplication
+			! style?.border?.style
+		) {
+			return [
+				...output,
+				...borderStyleGenerate(
+					{ border: { style: 'solid' } },
+					options
+				),
+			];
+		}
+
+		return output;
+	},
```

Comment:
One interesting thing to note here is that the architecture of these rules is "too abstracted", Basically, I have to repeat the same fallback code because it's organized as if "one style property" = "one CSS rule output" and it shouldn't be because "style" is an abstraction and not just another way to write CSS.

Anyway, I managed to make it work and the good thing is that for the frontend at least, I had to update only the style engine to make the fallback work for both global styles and block supports.